ext/pcntl/pcntl.c

```diff
@@ -385,6 +400,39 @@ PHP_FUNCTION(pcntl_waitpid)
 }
 /* }}} */
 
+#if defined (HAVE_WAITID) && defined (HAVE_POSIX_IDTYPES) && defined (HAVE_DECL_WEXITED) && HAVE_DECL_WEXITED == 1
+PHP_FUNCTION(pcntl_waitid)
+{
+	zend_long idtype = P_ALL;
+	zend_long id = 0;
+	bool id_is_null = 1;
+	zval *user_siginfo = NULL;
+	zend_long options = WEXITED;
+
+	ZEND_PARSE_PARAMETERS_START(0, 4)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_LONG(idtype)
+		Z_PARAM_LONG_OR_NULL(id, id_is_null)
```

Comment:
why is this parameter nullable? Apparently, it's not used anywhere.

Reply:
It's a minor style point. From the API design standpoint it's not strictly necessary to allow `id = null`, but since the `id` parameter will be ignored in the characteristic case of `idtype = P_ALL`, it can allow PHP programmers to better express their intent and help readability.

For example, compare:

```php
pcntl_waitid(P_ALL, null, $info);
```

with

```php
pcntl_waitid(P_ALL, 0, $info);
```

or even

```
```php
pcntl_waitid(P_ALL, 42, $info);
```

They all do the same thing, but IMHO the first version best communicates to the person reading the PHP code that we don't really care about the value of the second param (`id`).

In the C code, the `id_is_null` value is not used anywhere, as passing 0 to the system call will suffice, but I'm not sure if it's possible to allow the value to be null without using this Z_PARAM_LONG_OR_NULL macro.